hal/commands/login-to-ecr.yml

```
@@ -18,12 +23,30 @@ steps:
   - run:
       name: Install jq
       command: |
-          sudo apt-get update
-          sudo apt-get install jq -y
+          if command -v jq >> /dev/null 2>&1; then
+              echo "jq is already installed..."
+          else
+              if command -v apt-get >> /dev/null 2>&1; then
+                  sudo_me="sudo "
+                  if [ "$(whoami)" == "root" ] ; then
+                    sudo_me=""
+                  fi
+
+                  ${sudo_me} apt-get update
+                  ${sudo_me} apt-get install jq -y
+              else
+                  echo "jq is required. Please install it before running this step."
+                  exit 1
+              fi
+          fi
 
   - run:
       name: 'Generates Docker Credentials'
       command: |
+          if [[ "${<< parameters.api-token-var >>}" != "HAL_API_TOKEN" ]] ; then
+              echo "export HAL_API_TOKEN=${<< parameters.api-token-var >>}" >> $BASH_ENV
```

Sorry I missed up here. My original example I used because it was set for the next job. Since this var is used within the same bash shell it can just be 
```suggestion
              export HAL_API_TOKEN="${<< parameters.api-token-var >>}"
```